### PR TITLE
Fixed `Get-AzRoleAssignment` BadRequest when scope is '/' [#20323]

### PR DESCRIPTION
**Pull request by rachelxj-ms**
## Description

Fixed Get-AzRoleAssignment BadRequest when scope is '/'
reference: https://learn.microsoft.com/en-us/azure/role-based-access-control/role-assignments-list-rest
<html>
<body>

Filter | Description
-- | --
$filter=atScope() | Lists role assignments for only the specified scope, not including the role assignments at subscopes.
$filter=assignedTo('{objectId}') | Lists role assignments for a specified user or service principal.If the user is a member of a group that has a role assignment, that role assignment is also listed. This filter is transitive for groups which means that if the user is a member of a group and that group is a member of another group that has a role assignment, that role assignment is also listed.This filter only accepts an object ID for a user or a service principal. You cannot pass an object ID for a group.
$filter=atScope()+and+assignedTo('{objectId}') | Lists role assignments for the specified user or service principal and at the specified scope.
$filter=principalId+eq+'{objectId}' | Lists role assignments for a specified user, group, or service principal.

</body>
</html>
## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Resources/Resources/Models.Authorization/AuthorizationClient.cs [hunk 1]
```diff
@@ -209,15 +210,35 @@ public List<PSRoleAssignment> FilterRoleAssignments(FilterRoleAssignmentsOptions
...
                     principalId = string.IsNullOrEmpty(options.ADObjectFilter.Id) ? adObject.Id.ToString() : options.ADObjectFilter.Id;
-                    odataQuery = new ODataQuery<RoleAssignmentFilter>(f => f.PrincipalId == principalId);
+                    if (!hasScope)
+                    {
+                        odataQuery = new ODataQuery<RoleAssignmentFilter>(f => f.PrincipalId == principalId);
+                    } 
+                    else
+                    {
+                        // https://learn.microsoft.com/en-us/azure/role-based-access-control/role-assignments-list-rest
+                        odataQuery = new ODataQuery<RoleAssignmentFilter>(f => f.AtScope() && f.AssignedTo(principalId));
```
Comment (BethanyZhou) [inline src/Resources/Resources/Models.Authorization/AuthorizationClient.cs:232]: Previous query:
//subscriptions/4004a9fd-d58e-48dc-aeb2-4a4aec58606f/resourceGroups/PowershellTest/providers/Microsoft.Authorization/roleAssignments?$filter=principalId%20eq%20'f2dc21ac-702a-4bde-a4ce-146edf751d81'&api-version=2020-08-01-preview",
Current query:
Unable to find a matching HTTP request for URL 'GET //subscriptions/4004a9fd-d58e-48dc-aeb2-4a4aec58606f/resourceGroups/PowershellTest/providers/Microsoft.Authorization/roleAssignments?$filter=atScope()%20and%20assignedTo('f2dc21ac-702a-4bde-a4ce-146edf751d81')'. Calling method Item().

Please check this line whether is correct way to write query.

**Review by BethanyZhou — CHANGES_REQUESTED** [review]
Is it a reasonable scenario to fill scope as '/'?

File: src/Resources/Resources/Models.Authorization/AuthorizationClient.cs (hunk 1 above)
Comment (rachelxj-ms) [reply src/Resources/Resources/Models.Authorization/AuthorizationClient.cs:232]: It's resolved, filter condition uses '+' as separator

**Comment by rachelxj-ms** [comment]
close it for now as it requires to record new test.

**Pull request closed without merging**